ui/cypress/support/commands.js

```diff
@@ -10,7 +10,44 @@
 //
 //
 // -- This is a parent command --
-// Cypress.Commands.add('login', (email, password) => { ... })
+Cypress.Commands.add('login', (username, password = '') => {
+  // cy.login will be called inside beforeEach,
+  // cy.session stores cookies and localStorage when user first login,
+  // the cookies and localStorage will be reused in the feature beforeEach test.
+  cy.session(
+    [username, password],
+    () => {
+      // root login
+      cy.visit('/')
+      cy.get('[data-e2e=signin_submit]').click()
+      const dashboardAuthToken = localStorage.getItem('dashboard_auth_token')
+      window.localStorage.setItem('dashboard_auth_token', dashboardAuthToken)
```

Comment:
Are `localStorage` and `window.localStorage` different?

Reply:
There is no difference between the window.localStorage and localStorage in Cypress.